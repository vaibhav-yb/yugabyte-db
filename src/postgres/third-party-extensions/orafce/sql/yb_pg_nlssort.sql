-- Tests for nlssort
-- YB: we do not support SQL_ASCII encoding
\set ECHO none
SET client_min_messages = error;
DROP DATABASE IF EXISTS regression_sort;
CREATE DATABASE regression_sort WITH TEMPLATE = template0 ENCODING='UTF8' LC_COLLATE='C' LC_CTYPE='en_US.UTF-8';
\c regression_sort
SET client_min_messages = error;
CREATE EXTENSION orafce;
SET client_min_messages = default;
CREATE TABLE test_sort (name TEXT);
INSERT INTO test_sort VALUES ('red'), ('brown'), ('yellow'), ('Purple');
SELECT * FROM test_sort ORDER BY NLSSORT(name, 'C');
SELECT * FROM test_sort ORDER BY NLSSORT(name, '');
SELECT set_nls_sort('invalid');
SELECT * FROM test_sort ORDER BY NLSSORT(name);
SELECT set_nls_sort('');
SELECT * FROM test_sort ORDER BY NLSSORT(name);