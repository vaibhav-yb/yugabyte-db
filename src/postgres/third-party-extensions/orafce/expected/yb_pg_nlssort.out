-- Tests for nlssort
-- YB: we do not support SQL_ASCII encoding
\set ECHO none
  name
--------
 Purple
 brown
 red
 yellow
(4 rows)

  name
--------
 Purple
 brown
 red
 yellow
(4 rows)

 set_nls_sort
--------------

(1 row)

ERROR:  failed to set the requested LC_COLLATE value [invalid]
CONTEXT:  SQL function "nlssort" statement 1
 set_nls_sort
--------------

(1 row)

  name
--------
 Purple
 brown
 red
 yellow
(4 rows)

WARNING:  there is no transaction in progress
